android/src/main/java/com/oktareactnative/OktaSdkBridgeModule.java

```diff
@@ -673,14 +676,23 @@ private int getRecalculatedRequestCodeForActivityResult(
         if (resultCode == Activity.RESULT_OK) {
             final String callbackUri = data != null ? data.getDataString() : "";
             if (callbackUri == null) return initialRequestCode;
-            if (callbackUri.startsWith(this.config.getRedirectUri().toString())) {
+            if (LastRequestType.SIGN_IN == mLastRequestType &&
```

Comment:
@NikitaAvraimov-okta Does it make sense to validate callback uri on JS level to keep the same behavior on iOS and Android?

Reply:
Not really, that`s an android feature to corrupt activity request codes. As far as I know iOS worked fine and only android changes affected this behavior.